packages/react-client/src/types.ts

```diff
@@ -178,6 +178,8 @@ export type Device = {
 
 export type AudioDevice = Device & { isAudioPlaying: boolean };
 
+export type ToggleMode = "soft" | "hard";
```

Comment:
I believe we could come up with a better name. Here are some other suggestions:

1. Just a boolean: `toggle(forceStop: boolean = true)` 
2. `toggle(mode: "suspend" | "turnOff" = "turnOff")`
3. `toggle(onStop: "disable" | "stop" = "stop")`

However, I’m not satisfied with any of these options. Please help!

Reply:
i am more of a fan of implementing two functions like
`toggleDevice`
and 
`toggleStreaming`.
we get similar logic and we don't need to pass an argument so the toggle function can be put as an event handler

Reply:
I like that approach; it has a few advantages as well as disadvantages:

## Advantages
- Greater granularity
- Predictable behaviour
- plus everything you just mentioned

## Disadvantages
- Two functions instead of one; the user has to invoke both of them.
- On mobile, we don't have such separation; maybe we should add them there.
- To be compatible with the approach that I proposed, we need to either disable the stream in `toggleStreaming` to achieve an immediate `mute` effect (which is strange, because we just stopped streaming) or add separate `toggleMute / toggleDisable` functions, which would introduce another `toggle` function 😕

So for now, I want to merge this single powerful function